code/game/objects/effects/spawners/unique_spawner.dm

```diff
@@ -0,0 +1,129 @@
+/***
+ * How to use:
+ * Either place the base unique_spawner in the map and define the key_override to determine which spawners belong to one another.
+ * Or make a child instance of the unique_spawner and define the values there.
+ * When using the base version be sure to define both total_amount and max_per_spawner on only one of the spawners
+***/
+
+/obj/effect/spawner/unique_spawner
+	name = "unique spawner"
+	icon = 'icons/mob/screen_gen.dmi'
+	icon_state = "x2"
```

Comment:
It currently uses the default icon. I ain't a mapper so I don't know what is handy here